astronomy/orbital_elements_test.cpp

```diff
@@ -0,0 +1,422 @@
+  }
+
+  // Completes |initial_osculating_elements| and returns a GCRS trajectory
+  // obtained by flowing the corresponding initial conditions in |ephemeris|.
+  static not_null<std::unique_ptr<DiscreteTrajectory<GCRS>>>
+  EarthCentredTrajectory(
+      KeplerianElements<GCRS>& initial_osculating_elements,
+      Instant const& initial_time,
+      Instant const& final_time,
+      Ephemeris<ICRS>& ephemeris) {
+    MassiveBody const& earth = [&ephemeris]() -> MassiveBody const& {
+      for (not_null<MassiveBody const*> const body : ephemeris.bodies()) {
+        if (body->name() == "Earth") {
+          return *body;
+        }
+      }
+      LOG(FATAL) << "Ephemeris has no Earth";
+    }();
+    ephemeris.Prolong(final_time);
+    BodyCentredNonRotatingDynamicFrame<ICRS, GCRS> gcrs{&ephemeris, &earth};
+    DiscreteTrajectory<ICRS> icrs_trajectory;
+    KeplerOrbit<GCRS> initial_osculating_orbit{earth,
+                                               MasslessBody{},
+                                               initial_osculating_elements,
+                                               initial_time};
+    initial_osculating_elements = initial_osculating_orbit.elements_at_epoch();
+    icrs_trajectory.Append(
+        initial_time,
+        gcrs.FromThisFrameAtTime(initial_time)(
+            DegreesOfFreedom<GCRS>{GCRS::origin, Velocity<GCRS>{}} +
+            initial_osculating_orbit.StateVectors(initial_time)));
+    ephemeris.FlowWithAdaptiveStep(
+        &icrs_trajectory,
+        Ephemeris<ICRS>::NoIntrinsicAcceleration,
+        final_time,
+        Ephemeris<ICRS>::AdaptiveStepParameters{
+            EmbeddedExplicitRungeKuttaNyströmIntegrator<
+                DormandالمكاوىPrince1986RKN434FM,
+                Position<ICRS>>(),
+            /*max_steps=*/std::numeric_limits<std::int64_t>::max(),
+            /*length_integration_tolerance=*/1 * Milli(Metre),
+            /*speed_integration_tolerance=*/1 * Milli(Metre) / Second
+        },
+        /*max_ephemeris_steps=*/std::numeric_limits<std::int64_t>::max(),
+        /*last_point_only=*/false);
+    auto result = make_not_null_unique<DiscreteTrajectory<GCRS>>();
+    for (auto it = icrs_trajectory.Begin(); it != icrs_trajectory.End(); ++it) {
+      result->Append(
+          it.time(),
+          gcrs.ToThisFrameAtTime(it.time())(it.degrees_of_freedom()));
+    }
+    return result;
+  }
+
+  SolarSystem<ICRS> solar_system_;
+
+  // The solar system.
+  not_null<std::unique_ptr<Ephemeris<ICRS>>> real_ephemeris_;
+  MassiveBody const& real_earth_;
+
+  // An oblate Earth with no other bodies.
+  not_null<std::unique_ptr<Ephemeris<ICRS>>> oblate_earth_ephemeris_;
+  OblateBody<ICRS> const& oblate_earth_;
+
+  // A spherical Earth with no other bodies.
+  not_null<std::unique_ptr<Ephemeris<ICRS>>> spherical_earth_ephemeris_;
+  MassiveBody const& spherical_earth_;
+};
+
+
+TEST_F(OrbitalElementsTest, KeplerOrbit) {
```

Comment:
I think it would be better to construct the ephemerides in each test.  Apparently each ephemeris is only used in one test.  That would make the construction and the usage closer to one another and the test fixtures smaller.

Reply:
Done.